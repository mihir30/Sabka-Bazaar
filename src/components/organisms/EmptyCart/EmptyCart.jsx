import React from "react";
import { Button } from "../../atoms";
import "./EmptyCart.scss";
const EmptyCart = ({ closeBtnMethod, goToProducts }) => {
  return (
    <div className="minicart-modal-background">
      <div className="minicart-empty">
        <div className="cart-quantity-tile" >
          <p>
            <strong data-testid="my-cart-text">My Cart </strong>
          </p>
        </div>

        <div className="cart-quantity-tile-modal"  >
          <p>
            <strong data-testid="my-cart-text-modal">My Cart </strong>
          </p>

          <i
            id="close-btn-empty-cart"
            className="fas fa-times"
            onClick={closeBtnMethod}
          ></i>
        </div>

        <div>
          <p className="empty-cart-text" data-testid="no-items">No items in your cart</p>
          <p className="empty-cart-desc" data-testid="fav-items">
            Your favourite items are just a click away
          </p>
        </div>

        <div className="checkout-empty-tile">
          <div>
            <Button data-testid="start-shopping-button" buttonType={"checkout"} method={goToProducts}>
              <span className="start-shopping-btn">
                <span>Start Shopping</span>
              </span>
            </Button>
          </div>
        </div>
      </div>
    </div>
  );
};
export default EmptyCart;
