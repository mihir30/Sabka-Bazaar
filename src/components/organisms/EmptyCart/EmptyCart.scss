$color-white: white;
$background-color-white: white;
$background-color-pink: #cc0044;

.checkout-empty-tile {
	display: block;
	background-color: $background-color-white;
	position: relative;
	margin: 300px 0 0 0;
}

.minicart-empty {
	margin: 5px 0px 0px 0px;
	background-color: $background-color-white;
}

.empty-cart-text {
	font-weight: bold;
	text-align: center;
	margin-top: 100px;
}

.empty-cart-desc {
	text-align: center;
}

@media only screen and (min-width: 768px) {
	.checkout-empty-tile {
		display: block;
		background-color: $background-color-white;
		position: relative;
		margin: 680px 0 0 0;
	}

	.empty-cart-text {
		font-weight: bold;
		text-align: center;
		margin-top: 200px;
	}

	.empty-cart-desc {
		text-align: center;
	}
}

@media only screen and (min-width: 1024px) {
	.checkout-empty-tile {
		display: block;
		background-color: $background-color-white;
		position: relative;
		margin: 200px 0 0 0;
	}

	.start-shopping-btn {
		background-color: $background-color-pink;
		width: 500px;
		border-radius: 2px;
		margin: 15px 5px 10px 5px;
		color: $color-white;
		border: 2px solid background-color-pink;
		font-size: 18px;
		padding: 10px 50px;
	}

	#close-btn-empty-cart {
		font-size: 1.5rem;
		float: right;
		margin: 0px 5px 0px 315px;
		color: $color-white;
		cursor: pointer;
	}
}