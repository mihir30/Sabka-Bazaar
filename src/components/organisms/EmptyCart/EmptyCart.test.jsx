import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import EmptyCart from '../../organisms/EmptyCart/EmptyCart'

describe("Empty Cart Component", () => {
    it('renders the my cart text component properly', () => {
        render(<BrowserRouter><EmptyCart /></BrowserRouter>);
        expect(screen.getByTestId("my-cart-text")).toHaveTextContent('My Cart');
    });

    it('renders the my cart text modal component properly', () => {
        render(<BrowserRouter><EmptyCart /></BrowserRouter>);
        expect(screen.getByTestId("my-cart-text-modal")).toHaveTextContent('My Cart');
    });

    it('renders no items in cart text component properly', () => {
        render(<BrowserRouter><EmptyCart /></BrowserRouter>);
        expect(screen.getByTestId("no-items")).toHaveTextContent('No items in your cart');
    });

    it('renders favourite items in cart text component properly', () => {
        render(<BrowserRouter><EmptyCart /></BrowserRouter>);
        expect(screen.getByTestId("fav-items")).toHaveTextContent('Your favourite items are just a click away');
    });

    it('should have the right empty cart component snapshot', () => {
        const tree = renderer.create(<BrowserRouter><EmptyCart /></BrowserRouter>).toJSON();
        expect(tree).toMatchSnapshot();
    })

});