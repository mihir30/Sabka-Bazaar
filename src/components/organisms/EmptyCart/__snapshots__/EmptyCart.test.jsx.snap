// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty Cart Component should have the right empty cart component snapshot 1`] = `
<div
  className="minicart-modal-background"
>
  <div
    className="minicart-empty"
  >
    <div
      className="cart-quantity-tile"
    >
      <p>
        <strong
          data-testid="my-cart-text"
        >
          My Cart 
        </strong>
      </p>
    </div>
    <div
      className="cart-quantity-tile-modal"
    >
      <p>
        <strong
          data-testid="my-cart-text-modal"
        >
          My Cart 
        </strong>
      </p>
      <i
        className="fas fa-times"
        id="close-btn-empty-cart"
      />
    </div>
    <div>
      <p
        className="empty-cart-text"
        data-testid="no-items"
      >
        No items in your cart
      </p>
      <p
        className="empty-cart-desc"
        data-testid="fav-items"
      >
        Your favourite items are just a click away
      </p>
    </div>
    <div
      className="checkout-empty-tile"
    >
      <div>
        <button
          className="btn-checkout"
        >
          <span
            className="start-shopping-btn"
          >
            <span>
              Start Shopping
            </span>
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
